File: functions/libgcrypt/variants/go/1_ctz_5.go
package main

func ctz(in uint64) int32 {
	for i := int32(0); i < 64; i++ {
		if (in>>uint(i))&1 == 1 {
			return i
		}
	}
	return 64
}


func main(){
    ctz(0)
}

File: functions/libgcrypt/variants/c/1_ctz_5.c
#include <stdint.h>

int ctz(uint64_t in)
{
 int r = 0;
 for (uint64_t m = 1; m != 0; m <<= 1, r++)
 {
  if ((in & m) != 0)
   break;
 }
 return r;
}


int main(){
    ctz(0);
}